### PR TITLE
Start reworking Airbyte CI README: remove legacy connector_ops pieces

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What's going on

I'm trying to figure out what are the highest leverage pieces of developer tools that we can improve to unblock our productivity. The first step is to take stock of what we have today, and how it works. 

As I read through code and unearth pieces that are not used, I clean them up, inspired by @postamar's example ;-)

## This PR

This commit starts the root-level Airbyte CI readme with the list of tools:
- Add the readme table of subprojects
- Rework `connector_ops` readme
- Remove legacy checks that `connector_ops` had that are not used
- Remove legacy publish and test flows for connectors

### CODE REVIEW

**Comment by natikgadzhi** [comment]
Oh lol, `ci_credentials` seems to want a new poetry.lock, but the irony is, I didn't touch `pyproject.toml` there.

File: airbyte-ci/README.md [hunk 1]
```diff
@@ -2,4 +2,13 @@
 
 This folder is a collection of systems, tools and scripts that are used to run Airbyte's CI/CD
 
-The installation instructions for the `airbyte-ci` CLI tool cal be found here [airbyte-ci/connectors/pipelines](connectors/pipelines/README.md)
\ No newline at end of file
+The installation instructions for the `airbyte-ci` CLI tool cal be found here
```
Comment (alafanechere) [inline airbyte-ci/README.md:5]:
```suggestion
The installation instructions for the `airbyte-ci` CLI tool can be found here
```

File: airbyte-ci/README.md [hunk 2]
```diff
@@ -2,4 +2,13 @@
...
+[airbyte-ci/connectors/pipelines](connectors/pipelines/README.md)
+
+## Tools
+
+| Directory                                     | Description                                                        |
+| --------------------------------------------- | ------------------------------------------------------------------ |
+| [`base_images`](connectors/base_images)       | A set of tools to build and publish Airbyte base connector images. |
+| [`ci_credentials`](connectors/ci_credentials) | A CLI tool to fetch connector secrets from GCP Secrets Manager.    |
+| [`connector_ops`](connectors/connector_ops)   | A set of tools to verify connectors in CI.                         |
```
Comment (alafanechere) [inline airbyte-ci/README.md:14]:
```suggestion
| [`connector_ops`](connectors/connector_ops)   | A python package with utils reused in internal packages.                         |
```

File: airbyte-ci/connectors/ci_credentials/README.md [hunk 3]
```diff
@@ -68,31 +69,29 @@ pipx install --editable airbyte-ci/connectors/ci_credentials/
 
 This is useful when you are making changes to the package and want to test them in real-time.
 
-Note:
-
-- The package name is `pipelines`, not `airbyte-ci`. You will need this when uninstalling or reinstalling.
-- Even with the above `--editable` method, live changes to the code in the sibling project `/airbyte-ci/connectors/connector_ops/` are not automatically captured. To ensure you are using the latest code, use the command `pipx reinstall pipelines`.
+> [!Note]
+> - The package name is `ci_credentials`, not `airbyte-ci`. You will need this when uninstalling or reinstalling.
+> - Even with the above `--editable` method, live changes to the code in the sibling project `/airbyte-ci/connectors/connector_ops/` are not automatically captured. To ensure you are using the latest code, use the command `pipx reinstall ci_credentials`.
```
Comment (alafanechere) [inline airbyte-ci/connectors/ci_credentials/README.md:74]: `ci_credentials` does not depend on on `connector_ops`, why would we want to capture changes on `connector_ops`?

File: airbyte-ci/connectors/ci_credentials/README.md [hunk 4]
```diff
@@ -27,23 +28,22 @@ python -m pip install --user pipx
...
-_Note: `--force` is required to ensure updates are applied on subsequent installs._
-_Note: `--python=python3.10` is required to ensure the correct python version is used._
-_Note: `--editable` is required to ensure the correct python version is used._
-
-If you face any installation problem feel free to reach out the Airbyte Connectors Operations team.
-
+> [!Note]
+> - `--force` is required to ensure updates are applied on subsequent installs.
+> - `--python=python3.10` is required to ensure the correct python version is used.
+> - `--editable` is required to ensure the correct python version is used.
```
Comment (alafanechere) [inline airbyte-ci/connectors/ci_credentials/README.md:42]: `- `--editable` is required to ensure the correct python version is used.`

This sounds weird to me. Editable ensures that changes to the package code do not require a re-install to be captured, it should not have anything to do with the python version.

File: airbyte-ci/connectors/connector_ops/README.md
Comment (alafanechere) [inline airbyte-ci/connectors/connector_ops/README.md:1]: @natikgadzhi FYI I'm not satisfied with the existence of this package as:
* Its name relates to a team which does not exists anymore
* Its scope is not clear at all. We stuffed a lot of different things in there: models and light CLI used in CI.
* Its mainly used currently to expose the useful `Connector` class, which we could move to a different package with a more explicit name.

I'm responsible for this situation as I orignally introduced this package 😄

**Review by alafanechere — APPROVED** [review]
Thanks for the effort!
Minor suggestions.

File: airbyte-ci/connectors/ci_credentials/README.md (hunk 3 above)
Comment (natikgadzhi) [reply airbyte-ci/connectors/ci_credentials/README.md:74]: I have little idea of what was about tbh. Let me cut that out then.

File: airbyte-ci/connectors/connector_ops/README.md
Comment (natikgadzhi) [reply airbyte-ci/connectors/connector_ops/README.md:1]: Well that is the next question — please capture this in an issue and let's retire the tools and move `Connector` into utils or something. I'm open to this.

File: airbyte-ci/connectors/ci_credentials/README.md (hunk 4 above)
Comment (natikgadzhi) [inline airbyte-ci/connectors/ci_credentials/README.md:42]:
```suggestion
```

File: airbyte-ci/connectors/ci_credentials/README.md (hunk 3 above)
Comment (natikgadzhi) [inline airbyte-ci/connectors/ci_credentials/README.md:74]:
```suggestion
```

**Pull request merged**